server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -226,6 +227,17 @@ public RestoreService(
      */
     public void restoreSnapshot(final RestoreSnapshotRequest request, final ActionListener<RestoreCompletionResponse> listener) {
         try {
+            // INDEX_STORE_TYPE_SETTING is intended to be a system-managed index setting that is configured when restoring a snapshot and
+            // should not be set to value REMOTE_SNAPSHOT by user.
+            if (request.indexSettings()
```

Comment:
It would be better if we block setting the property on the request irrespective of the value being set. The property is system managed and should never be set by the user.

Reply:
We still support setting the property `index.store.type` to any storage type, more info in [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/index-modules-store.html). So we have to block it on `remote_snapshot` value specifically and not other values.

One other thing I observed, is property `index.store.type` can also be set while creating a new index. The setting `index.store.type` is expert level setting, most users would stick to default but still there is an edge case here. So to block user completely from setting property `index.store.type` to `remote_snaphot`, we have to block (throw relevant error with message) it from both restore path (which is done by this PR) and also index creation path which can be done in separate PR.

Reply:
opened #11867